src/Model/Ticket.java

```diff
@@ -3,32 +3,30 @@
 import constants.ConcertHall;
 import constants.Formatters;
 import constants.StadiumSector;
+import utils.NullableWarning;
+import utils.NullableWarningProcessor;
 
 import java.math.BigDecimal;
 import java.math.RoundingMode;
 import java.time.LocalDateTime;
+import java.util.Objects;
 
-public class Ticket {
-    private static int counter = 0;
-
-    private int id;
+public class Ticket extends IdentifiableEntity implements Printable, Sharable {
     private ConcertHall concertHall;
+    @NullableWarning
     private String eventCode;
+    @NullableWarning
     private LocalDateTime time;
     private final LocalDateTime creationTime;
-    private boolean isPromo;
+    private boolean isPromo = false;
+    @NullableWarning
     private StadiumSector stadiumSector;
     private BigDecimal maxBackpackWeight;
     private BigDecimal price;
 
-
-    private void initID(){
-        this.id = counter;
-        counter++;
-    }
-
     public Ticket() {
-        initID();
+        generateID();
+        NullableWarningProcessor.processAnnotation(this);
```

Comment:
the logic with annotation should be implemented in a way you don't need to call 'processAnnotation' or smth else